### ISSUE TITLE
Unify linear & quadratic models

### ISSUE THREAD

**Issue by tmcclintock** [Bramson-Planetary-Research-Group/Mars-Troughs#29]
Right now we have a [linear accumulation model](https://github.com/Bramson-Planetary-Research-Group/Mars-Troughs/blob/master/mars_troughs/accumulation_model.py#L70) and a [linear lag model](https://github.com/Bramson-Planetary-Research-Group/Mars-Troughs/blob/master/mars_troughs/lag_model.py#L54). This is a bit clunky, because the linear function itself is being repeatedly needlessly. As in, the only "real" difference between these models is what the "x" and "y" values are (and really, we *could* unify all the polynomial models into one, but that can come after this issue).

This issue proposes unifying these different linear models by creating a single generic `LinearModel` class:

```python
# linear model.py or maybe in model.py

from typing import List

class LinearModel(Model):
    # Generic arbitrary initial values
    intercept: float = 1.0
    slope: float = 1.0

    @property
    def parameter_names(self) -> List[str]:
        return ["intercept", "slope"]

    def eval(self, x):
        return self.intercept + x * self.slope
```

Now in the linear accumulation/lag models we can use this as a *mixin* class 

```python
# lag_model.py
# a similar thing will exist in accumulation_model.py

class LinearLag(LagModel, LinearModel):

    def get_lag_at_t(self, time):
        return self.eval(time)
```

This could be extended (in a future issue, if we want) to include generic `QuadraticModel` and `ConstantModel` classes, or we could jump straight to a fully generic `PolynomialModel` that can handle every degree.

**Comment by tmcclintock** [comment]
**Update** - tagging @klaferr 

In the proposal above I didn't realize that the *class variables* (e.g. `slope` and `intercept`) were not automatically set at construction, requiring us to still have `__init__` functions where they are set. The solution is to turn the class variables in to regular instance variables, and when necessary refer to the `LinearModel.__init__` function:
```python
# linear_model.py

class LinearModel(Model):

    def __init__(self, intercept: float = 1.0, slope: float = 1.0):
        # Generic arbitrary initial values
        self.intercept = intercept
        self.slope = slope

    ...
```

```python
# accumulation_model.py

class LinearInsolationAccumulation(InsolationAccumulationModel, LinearModel):
    def __init__(
        self,
        times: np.ndarray,
        insolations: np.ndarray,
        intercept: float = 1e-6,
        slope: float = 1e-6,
    ):
        # 'super' refers to the InsolationAccumulationModel
        super().__init__(times, insolations)
        # We also need to call the LinearModel constructor
        LinearModel.__init__(self, intercept, slope)
```

The same change would be needed in `lag_model.py`.

**Comment by tmcclintock** [comment]
Closed with #30
